pkg/sources/reconciler/awssnssource/subscription.go

```diff
@@ -129,8 +129,8 @@ func (r *Reconciler) ensureUnsubscribed(ctx context.Context) error {
 
 	src := v1alpha1.SourceFromContext(ctx)
 
-	url := src.GetStatusManager().Address.URL
-	if url == nil {
+	sm := src.GetStatusManager()
+	if sm == nil || sm.Address == nil || sm.Address.URL == nil {
```

Comment:
`sm` will never be nil, in case you want to shorten that conditional a little bit:

https://github.com/triggermesh/triggermesh/blob/bd69609b0c5d45a2a5c71f72239bc73d6e6904c5/pkg/apis/sources/v1alpha1/awssns_lifecycle.go#L53